### ISSUE TITLE
Disallow string concatenation when using __dirname and __filename (no-path-concat)

### ISSUE THREAD

**Issue by feross** [feross/standard#403]
See: http://eslint.org/docs/2.0.0/rules/no-path-concat

Replace this:

```js
var fullPath = __dirname + "/foo.js";
```

With this:

```js
var fullPath = path.join(__dirname, "foo.js");
```

From eslint docs:

> However, there are a few problems with this. First, you can't be sure what type of system the script is running on. Node.js can be run on any computer, including Windows, which uses a different path separator. It's very easy, therefore, to create an invalid path using string concatenation and assuming Unix-style separators. There's also the possibility of having double separators, or otherwise ending up with an invalid path.

> In order to avoid any confusion as to how to create the correct path, Node.js provides the path module. This module uses system-specific information to always return the correct value.

**Comment by feross** [comment]
```
# tests 427
# pass  395
# fail  32
```

7.5% of packages either don't work on Windows, or their tests don't work on Windows.

I think we should enable this, because even though this will make people change their code, this is actually a real problem – a programmer error — in their code that should be fixed.

**Comment by feross** [comment]
After fixing up all the packages that I have push access to, only 21/427 (5%) are failing. All the cases I fixed up were real cases where that code wouldn't work in a cross-platform way on Windows.

I think this is a good rule.

**Comment by jprichardson** [comment]
ACK

**Comment by feross** [comment]
Released in standard 6.0.0.
